File: src/game/rooms.ts
export interface Room {
  id: number
  name: string
  catGroup: 'basic' | 'rare' | 'special' | 'superhero' | 'ai'
  day: string
  night: string
  match3Background: string
}

export const rooms: Room[] = [
  { id: 1, name: 'Комната', catGroup: 'basic', day: '/assets/backgrounds/room-1-day.png', night: '/assets/backgrounds/room-1-night.png', match3Background: '/assets/backgrounds/match3-room-1.png' },
  { id: 2, name: 'Уютный приют', catGroup: 'rare', day: '/assets/backgrounds/room-2-day.png', night: '/assets/backgrounds/room-2-night.png', match3Background: '/assets/backgrounds/match3-room-2.png' },
  { id: 3, name: 'Котодом', catGroup: 'special', day: '/assets/backgrounds/room-3-day.png', night: '/assets/backgrounds/room-3-night.png', match3Background: '/assets/backgrounds/match3-room-3.png' },
  { id: 4, name: 'Котоцентр', catGroup: 'superhero', day: '/assets/backgrounds/room-4-day.png', night: '/assets/backgrounds/room-4-night.png', match3Background: '/assets/backgrounds/match3-room-4.png' },
  { id: 5, name: 'Штаб Котификации', catGroup: 'ai', day: '/assets/backgrounds/room-5-day.png', night: '/assets/backgrounds/room-5-night.png', match3Background: '/assets/backgrounds/match3-room-5.png' },
]

File: android/game/src/main/kotlin/dev/psychocat/catclicker/game/data/Rooms.kt
package dev.psychocat.catclicker.game.data

/**
 * Port of src/game/rooms.ts. Image fields are drawable resource names (see tools/android/convert_assets.py),
 * so this module stays free of Android dependencies.
 */
data class Room(
    val id: Int,
    val name: String,
    val catGroup: String,
    val dayImage: String,
    val nightImage: String,
    val match3Image: String,
)

object Rooms {
    private fun room(id: Int, name: String, catGroup: String) = Room(
        id = id,
        name = name,
        catGroup = catGroup,
        dayImage = "room_${id}_day",
        nightImage = "room_${id}_night",
        match3Image = "match3_room_$id",
    )

    val all: List<Room> = listOf(
        room(1, "Комната", "basic"),
        room(2, "Уютный приют", "rare"),
        room(3, "Котодом", "special"),
        room(4, "Котоцентр", "superhero"),
        room(5, "Штаб Котификации", "ai"),
    )

    val count: Int get() = all.size

    fun byId(id: Int): Room = all[id - 1]
}
